File: algorithms/javascript/two-sum.js
function twoSum(nums,target){
    const indexMap = new Map()
    for (let i = 0; i < nums.length; i++) {
        if (indexMap.get(nums[i]) !== undefined) return [indexMap.get(nums[i]), i]
        indexMap.set(target - nums[i], i)
    }
    return []
}

console.log(twoSum([2, 7, 11, 15],9))

File: algorithms/go/two-sum.go
func twoSum(nums []int, target int) []int {
	indexMap := make(map[int]int, len(nums))

	for key, value := range nums {
		another := target - value
		if _key, ok := indexMap[another]; ok {
			return []int{_key, key}
		}
		indexMap[value] = key
	}
	return nil
}